scipy/stats/_distn_infrastructure.py

```diff
@@ -2164,16 +2173,34 @@ def restore(args, theta):
 
             def func(theta, x):
                 newtheta = restore(args[:], theta)
-                return self._penalized_nnlf(newtheta, x)
+                return objective(newtheta, x)
 
         return x0, func, restore, args
 
+    # TODO: no longer assume all variables are free
+    # TODO: order > len(theta)
+    # TODO: add tests
+    # TODO: don't re-calculate moments from data every iteration
+    def _moment_error_norm(self, theta, x):
+        loc, scale, args = self._unpack_loc_scale(theta)
+        if not self._argcheck(*args) or scale <= 0:
+            return inf
+        dist_moments = np.array([self.moment(n+1, *args, loc=loc, scale=scale)
+                                 for n in range(len(theta))])
+        # how can I import scipy.stats.moment?
+        # is there a function for moments NOT about mean?
```

Comment:
RE: this comment/question, not all distributions have them but `_munp` methods are non-central, 
https://github.com/scipy/scipy/blob/master/scipy/stats/_distn_infrastructure.py#L773-L776
the are centered around the number 0. 
Example here: 
https://github.com/scipy/scipy/blob/master/scipy/stats/_continuous_distns.py#L766

Reply:
Yes, and if `_munp` is not defined, then I suppose the `moment` method of any `rv_continuous` object falls back to a generic implementation for calculating non-central moments.  I was asking for a function to calculate non-central (sample) moments from _data_. I found it odd that `scipy.stats.moment` takes moments about the mean whereas `rv_continuous.moment` takes moments about zero.

Reply:
ah, ok, yeah that is a bit confusing, never thought about it before.

Reply:
Looks like `scipy.stats.moment` is substantially faster than calculating them manually, so I suppose everything could be based on central moments or we could convert the central moments to noncentral moments, but it doesn't really matter because these only get calculated once.

Reply:
No actual issue here, marking as resolved.